### PR TITLE
Bug 1670184 - Upgrade to rkv 0.15.0 and make the backend selectable

### PR DESCRIPTION
**Pull request by badboy**
The default is still Rkv's LMDB storage backend and we don't have any
intention to change that at the moment.

### CODE REVIEW

File: glean-core/src/database/mod.rs [hunk 1]
```diff
@@ -81,7 +110,12 @@ impl Database {
         let path = Path::new(data_path).join("db");
         log::debug!("Database path: {:?}", path.display());
 
+        // FIXME: This is probably more knowledge than we should have about the database internals.
+        // We could instead iterate over the directory and sum up the file sizes.
```
Comment (Dexterp37) [inline glean-core/src/database/mod.rs:114]: This seems like a reasonable fix. Would you kindly file a bug and add the number here?

File: glean-core/Cargo.toml [hunk 2]
```diff
@@ -41,3 +41,7 @@ env_logger = { version = "0.7.1", default-features = false, features = ["termcol
 tempfile = "3.1.0"
 iso8601 = "0.4"
 ctor = "0.1.12"
+
+[features]
+# Enable the "safe-mode" Rust storage backend instead of the default LMDB one.
+rkv-safe-mode = []
```
Comment (Dexterp37) [inline glean-core/Cargo.toml:47]: Does this mean the new glean-core version, once we cut a release, will use the new storage system? What happens with the legacy files? Do we need migration?

Comment (badboy) [reply glean-core/Cargo.toml:47]: No, it explicitly means we're staying on LMDB.

This feature allows us to toggle it on if we want to (which means we can more easily test this)

File: glean-core/src/database/mod.rs (hunk 1 above)
Comment (badboy) [reply glean-core/src/database/mod.rs:114]: yup, will do

**Review by Dexterp37 — APPROVED** [review]

**Pull request merged**